src/main.rs

```diff
@@ -32,8 +41,9 @@ fn main() {
             return Ok(Response::with((status::BadRequest, "Please send a BTSF file with precisely one chart in it")))
         }
 
-        let data_sets = req.get::<persistent::Read<DataSets>>().unwrap();
-        let result = lib::correlate::correlate(&input_charts[0], &data_sets[..]);
+        let mutex = req.get::<persistent::Write<CorrCache>>().unwrap();
+        let mut cache = mutex.lock().unwrap();
+        let result = cache.correlate(&input_charts[0], &DATA_SETS[..]);
 
```

Comment:
@trishume Can we somehow give up our lock on the mutex here? Since we don't need it anymore.